File: Python/page_object/07_sauce_labs/home_page.py
from search_page import SearchPage
from page_base import BasePage


class HomePage (BasePage):

    def search_for(self, keyword):
        search_field = self._driver.find_element_by_id("query")
        search_field.send_keys(keyword)
        search_field.submit()

        return SearchPage(self._driver)

File: Java/PageObject/07_SauceLabs/HomePage.java
package 07_SauceLabs;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import 07_SauceLabs.SearchPage;

public class HomePage extends PageBase{
		
	public HomePage(WebDriver driver) {
		this._driver = driver;
	}
	
	public SearchPage searchFor(String keyword){
        WebElement search_field = this._driver.findElement(By.id("query"));
        search_field.sendKeys(keyword);
        search_field.submit();

        return new SearchPage(this._driver);
	}
}
